vyper/builtin_functions/functions.py

```diff
@@ -178,21 +194,40 @@ class Convert(_BuiltinFunction):
     _id = "convert"
 
     def fetch_call_return(self, node):
+        self.infer_arg_types(node)
+        target_type = get_type_from_annotation(node.args[1], DataLocation.MEMORY)
+
+        # note: more type conversion validation happens in convert.py
+        return target_type
+
+    def infer_arg_types(self, node):
         validate_call_args(node, 2)
+
         target_type = get_type_from_annotation(node.args[1], DataLocation.MEMORY)
```

Comment:
i'm not convinced `DataLocation.MEMORY` is even kosher here. `convert` usually returns something on the stack.